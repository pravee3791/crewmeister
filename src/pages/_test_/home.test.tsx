import React from 'react';
import ReactDom from "react-dom";
import { BrowserRouter} from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from "../home";


afterEach(cleanup);

it('reders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render<any, any>(<BrowserRouter>
    <Home></Home>
    </BrowserRouter>, div)
})

it('checks if table exists', () => {
    const HomePage = render(<BrowserRouter><Home></Home></BrowserRouter>);
    const empTable = HomePage.container.querySelector('#employee-table');
    expect(empTable).toBeTruthy();
    expect(empTable).toBeDefined();
})

it(' Home Page matches snapshot', () => {
    const tree = renderer.create(<BrowserRouter><Home></Home></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})







