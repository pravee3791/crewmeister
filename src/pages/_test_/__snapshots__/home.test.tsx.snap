// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Componenet Page matches snapshot 1`] = `
<div
  className="home"
>
  <div
    className="home-table"
  >
    <div
      className="employee-header"
    >
      <div>
        Employee
      </div>
      <div>
        <button
          className="employee-submit"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </div>
    <div
      className="employee-container"
    >
      <div
        className="employee-row"
      >
        <div
          className="employee-label"
        >
          First Name
        </div>
        <div>
          <input
            className="employee-input"
            id="firstName"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="employee-row"
      >
        <div
          className="employee-label"
        >
          Last Name
        </div>
        <div>
          <input
            className="employee-input"
            id="lastName"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="employee-row"
      >
        <div
          className="employee-label"
        >
           Email: 
        </div>
        <div>
          <input
            className="employee-input"
            id="email"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="employee-row"
      >
        <div
          className="employee-label"
        >
          Gender
        </div>
        <div>
          <input
            className="employee-input"
            id="gender"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` Home Page matches snapshot 1`] = `
<div
  className="home"
>
  <div
    className="home-search"
  >
    <div
      className="search"
    >
      <div>
        <input
          className="search-input"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
      </div>
      <div>
        <a
          href="/create"
          onClick={[Function]}
        >
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span>
            Add employee
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    className="home-table"
    id="employee-table"
  >
    <table>
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Gender
          </th>
          <th>
            Status
          </th>
        </tr>
      </thead>
      <tbody
        id="employee-table-body"
      />
    </table>
  </div>
</div>
`;
